core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -2292,6 +2292,28 @@ private void checkLiteral2(String expression, String expected) {
         .ok(expectedDateMinusNegate);
   }
 
+  @Test public void testUnparseSqlIntervalQualifierBigQuery() {
+    final String sql0 = "select  * from \"employee\" where  \"hire_date\" - "
+            + "INTERVAL '19800' SECOND(5) > TIMESTAMP '2005-10-17 00:00:00' ";
+    final String expect0 = "SELECT *\n"
+            + "FROM foodmart.employee\n"
+            + "WHERE (hire_date - INTERVAL 19800 SECOND)"
+            + " > TIMESTAMP '2005-10-17 00:00:00'";
+    sql(sql0).withBigQuery().ok(expect0);
+
+    final String sql1 = "select  * from \"employee\" where  \"hire_date\" + "
+            + "INTERVAL '10' HOUR > TIMESTAMP '2005-10-17 00:00:00' ";
+    final String expect1 = "SELECT *\n"
+            + "FROM foodmart.employee\n"
+            + "WHERE (hire_date + INTERVAL 10 HOUR)"
+            + " > TIMESTAMP '2005-10-17 00:00:00'";
+    sql(sql1).withBigQuery().ok(expect1);
+
+    final String sql2 = "select  * from \"employee\" where  \"hire_date\" + "
+            + "INTERVAL '1 2:34:56.78' DAY TO SECOND > TIMESTAMP '2005-10-17 00:00:00' ";
+    sql(sql2).withBigQuery().throws_("Only INT64 is supported as the interval value for BigQuery.");
```

Comment:
Cannot add a range time unit test now as most likely range time unit has a non-long interval value.

Reply:
Then what is exactly the data type ?

Reply:
Sorry I don't understand your question. 

I meant because usually range time unit like "WEEK TO DAY" usually has non-int64 value, which will be reject by INT64 check before goes to time unit check.